pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -806,17 +806,14 @@ def select_cluster_environment(self) -> ClusterEnvironment:
         if self._cluster_environment is not None:
             return self._cluster_environment
         if self._is_slurm_managing_tasks():
-            env = SLURMEnvironment()
             rank_zero_info("Multiprocessing is handled by SLURM.")
-        elif TorchElasticEnvironment.is_using_torchelastic():
-            env = TorchElasticEnvironment()
-        elif KubeflowEnvironment.is_using_kubeflow():
-            env = KubeflowEnvironment()
-        elif LSFEnvironment.is_using_lsf():
-            env = LSFEnvironment()
-        else:
-            env = LightningEnvironment()
-        return env
+            return SLURMEnvironment()
+
+        for env_type in (TorchElasticEnvironment, KubeflowEnvironment, LSFEnvironment, LightningEnvironment):
```

Comment:
1. I don't think we need `LightningEnvironment` in this list. its inclusion will mean we never execute L816 because we will always return True because of `LightningEnvironment.detect()`'s implementation
2. Should we keep this list of non-LightningEnvironmen cluster environments as a constant somewhere? then if support for other clusters are added, we can update the registry once for wherever it's used
3. does the ordering of the list matter?

Reply:
1) agree, this can be done :) 
2) that has crossed my mind too. SLURM will be an outlier though. It's detection is slightly different and guarded through a method _is_slurm_managing_tasks in the accelerator connector.
3) Yes, TorchElasticEnv has priority over the Lightning because they share some env variable names. But we can take it out and make sure it comes last anyway, so the rest should not matter in ordering.